### ISSUE TITLE
what does this error mean? i indexed a big document

### ISSUE THREAD

**Issue by bryanhpchiang** [stanford-futuredata/ColBERT#165]
i indexed a big PDF (think 2000 pages)

getting this error

```
Traceback (most recent call last):
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 419, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/fastapi/applications.py", line 270, in __call__
    await super().__call__(scope, receive, send)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/starlette/middleware/cors.py", line 92, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/starlette/middleware/cors.py", line 147, in simple_response
    await self.app(scope, receive, send)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/starlette/routing.py", line 706, in __call__
    await route.handle(scope, receive, send)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/fastapi/routing.py", line 237, in app
    raw_response = await run_endpoint_function(
  File "/home/bryan/venv/gpu/lib/python3.10/site-packages/fastapi/routing.py", line 163, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/bryan/symbiotic/./main.py", line 214, in respond
    closest_ids, *_ = searcher.search(message, k=5)
  File "/home/bryan/ColBERT/colbert/searcher.py", line 61, in search
    return self.dense_search(Q, k, filter_fn=filter_fn)
  File "/home/bryan/ColBERT/colbert/searcher.py", line 108, in dense_search
    pids, scores = self.ranker.rank(self.config, Q, filter_fn=filter_fn)
  File "/home/bryan/ColBERT/colbert/search/index_storage.py", line 84, in rank
    scores, pids = self.score_pids(config, Q, pids, centroid_scores)
  File "/home/bryan/ColBERT/colbert/search/index_storage.py", line 130, in score_pids
    pids = pids[torch.topk(approx_scores, k=config.ndocs).indices]
RuntimeError: "topk_cpu" not implemented for 'Half'
```

**Comment by santhnm2** [comment]
Could you see if PR #166 fixes this?

**Comment by bryanhpchiang** [comment]
i want to but i'm also too lazy to merge it cuz i modified a bunch of stuff
i fixed the error by just moving approx_scores to the GPU so i presume your PR also works 

also did you TA 149 last qtr? lol

**Comment by santhnm2** [comment]
I see, that still helps thanks! - merged #166.
I did TA 149 last quarter. :)
